src/ScrollableContainers/_tk.py

```diff
@@ -32,27 +38,33 @@ def __init__(self, *args, **kwargs):
         # Scrollable canvas. This is the widget which actually manages
         # scrolling. Initially, it will be above the scrollbars, so the latter
         # won't be visible.
-        self._canvas = tk.Canvas(self)
+        self._canvas = tk.Canvas(self._frame)
         self._canvas.bind("<Configure>", self._on_canvas_configure)
         self._canvas.bind("<Enter>", self._on_canvas_enter)
         self._canvas.bind("<Leave>", self._on_canvas_leave)
         self._canvas.configure(xscrollcommand=self._xscrollbar.set, yscrollcommand=self._yscrollbar.set)
         self._canvas.grid(row=0, column=0, rowspan=2, columnspan=2, sticky=tk.NSEW)
 
-        self.grid_rowconfigure(0, weight=1)
-        self.grid_columnconfigure(0, weight=1)
+        self._frame.grid_rowconfigure(0, weight=1)
+        self._frame.grid_columnconfigure(0, weight=1)
 
-        self.frame = ttk.Frame(self._canvas)
-        self._window = self._canvas.create_window((0, 0), window=self.frame, anchor=tk.NW)
-        self.frame.bind("<Configure>", self._on_frame_configure)
-        self._on_frame_expose_id = self.frame.bind("<Expose>", self._on_frame_expose)
+        # The scrollable frame must be in the canvas for the latter to control
+        # its viewport. Hence, its parent widget is the canvas rather than what
+        # what was passed to this constructor.
+        super().__init__(self._canvas, *args, **kwargs)
+        self._window = self._canvas.create_window((0, 0), window=self, anchor=tk.NW)
+        self.bind("<Configure>", self._on_frame_configure)
+        self._on_frame_expose_id = self.bind("<Expose>", self._on_frame_expose)
 
         # Initially, the vertical scrollbar is a hair below its topmost
         # position. Move it to said position. No harm in doing the equivalent
         # for the horizontal scrollbar.
         self._canvas.xview_moveto(0.0)
         self._canvas.yview_moveto(0.0)
 
+    def pack(self, *args, **kwargs):
+        self._frame.pack(*args, **kwargs)
+
```

Comment:
This pattern will have to be repeated for every method which does something to `self` but should be done to `self._frame` instead! Consider scrapping this pull request.